tests/encoding_test.c

```diff
@@ -0,0 +1,368 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+
+#include <aws/common/encoding.h>
+#include <aws_test_harness.h>
+
+/* Test cases from rfc4648 for Base 16 Encoding */
+
+static int run_hex_encoding_test_case(struct aws_allocator *alloc, const char *test_str, size_t test_str_size,
+                                      const char *expected, size_t expected_size) {
+    size_t output_size = 0;
+
+    char *output_str = NULL;
+
+    ASSERT_SUCCESS(aws_hex_encode((uint8_t *)test_str, test_str_size - 1, NULL, &output_size), "encode call should have succeeded");
+    ASSERT_INT_EQUALS(expected_size, output_size, "Output size on string should be %d", expected_size);
+
+    output_str = (char *)aws_mem_acquire(alloc, output_size);
+    memset(output_str, 0, output_size);
+
+    ASSERT_SUCCESS(aws_hex_encode((uint8_t *)test_str, test_str_size - 1, output_str, &output_size), "encode call should have succeeded");
+    ASSERT_INT_EQUALS(expected_size, output_size, "Output size on an empty string should be %d", expected_size);
```

Comment:
... is `test_str` always going to be empty?

Reply:
no, that's left over from when this function was part of a particular test and I copy and pasted it during a refactor.